src/main/java/com/petros/bibernate/dao/EntityPersister.java

```diff
@@ -219,12 +261,50 @@ private static <T> T mapResultSetToEntity(Class<T> entityClass, ResultSet result
             for (var entityField : entityClass.getDeclaredFields()) {
                 entityField.setAccessible(TRUE);
                 String columnName = getColumnName(entityField);
-                entityField.set(entity, resultSet.getObject(columnName));
+                entityField.set(entity, convertToJavaType(entityField, resultSet.getObject(columnName)));
             }
             return entity;
         } catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException |
                  SQLException e) {
             throw new BibernateException(e);
         }
     }
+
+    private static Object convertToJavaType(Field field, Object value) throws IllegalAccessException, SQLException {
```

Comment:
We will move this method to separate class later. Hibernate has default convertors and allows to add our own.

Reply:
No objections.